src/globus_sdk/experimental/globus_app/_identity_id_reader.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+import typing as t
+
+from globus_sdk.experimental.tokenstorage import TokenStorage, TokenStorageData
+
+
+class IdentityIDReader:
```

Comment:
Still thinking about this class.

At least one thing that strikes me as odd off the bat: this isn't responsible for reading identity id from a response object, that functionality has stayed in the tokenstorage object.

I do still have a broader question in my head about whether this is the right type of thing to isolate in context, but if it is isolated it should probably own all of the reading

Reply:
This disappeared back into `ValidatingTokenStorage` in the last rewrites. I think we're in a pretty good place now.
I considered trying to define a reader which owns all of this functionality, but the `decode_id_token` case has a lot of potential subtlety to handle, so I'm not worrying about it for now.